crates/profiling/src/jitdump.rs

```diff
@@ -201,12 +201,12 @@ impl JitDumpAgent {
         match target_lexicon::HOST.architecture {
             Architecture::X86_64 => elf::header::EM_X86_64 as u32,
             Architecture::I686 => elf::header::EM_386 as u32,
-            Architecture::Arm => elf::header::EM_ARM as u32,
-            Architecture::Armv4t => elf::header::EM_ARM as u32,
-            Architecture::Armv5te => elf::header::EM_ARM as u32,
-            Architecture::Armv7 => elf::header::EM_ARM as u32,
-            Architecture::Armv7s => elf::header::EM_ARM as u32,
-            Architecture::Aarch64 => elf::header::EM_AARCH64 as u32,
+            Architecture::Arm(ArmArchitecture::Arm) => elf::header::EM_ARM as u32,
+            Architecture::Arm(ArmArchitecture::Armv4t) => elf::header::EM_ARM as u32,
+            Architecture::Arm(ArmArchitecture::Armv5te) => elf::header::EM_ARM as u32,
+            Architecture::Arm(ArmArchitecture::Armv7) => elf::header::EM_ARM as u32,
+            Architecture::Arm(ArmArchitecture::Armv7s) => elf::header::EM_ARM as u32,
```

Comment:
With the new API, instead of naming every subarchitecture here, we can just do `Architecture::Arm(_)` to match all ARM subarchitectures and map them to `EM_ARM` in a single go.

And ditto for AArch64 below.

Reply:
Ahh  .. ok. Thanks. Will update.